src/ChordSubs.hs

```diff
@@ -0,0 +1,272 @@
+module ChordSubs
+  ( remove5
+  , extend1
+  , negativeNote
+  , notesToChord
+  , negative
+  , dimFamilySub
+  , diatonicChord
+  , isDiatonicTo
+  , tonic
+  , supertonic
+  , mediant
+  , subdominant
+  , dominant
+  , submediant
+  , subtonic
+  , diatonicFuncSub
+  , parallelSub
+  , alteredDominantSub
+  ) where
+
+
+
+import Base.Chord
+import Base.Chord.Chord as C
+import Base.Chord.Extension
+import Base.Chord.HighestNatural
+import Base.Chord.Note
+import Base.Chord.Sus
+
+import Base.Core.Quality.CQuality as CQ
+import Base.Core.Quality.IQuality as IQ
+
+import Base.Interval hiding (getQuality)
+import qualified Base.Interval as I(getQuality)
+
+import Data.List(sortBy, delete, zip4, zip5, find, elemIndex)
+import Data.Function(on)
+import Lib(chordToIntervals, chordToNotes, qualityToIntervals)
+import Data.Set(Set(..), toList, member, isSubsetOf, fromList)
+import qualified Data.Set as S (delete)
+import Data.Maybe(fromJust, catMaybes, isJust)
+import Base.PitchClass(pitchClass)
+import qualified Data.Map.Strict as M (lookup, elems)
+import Common.Utils
+import Scale
+
+
+type ExpChord = (Chord, [Note])
+
+
+remove5 :: ExpChord -> ExpChord
+remove5 (chord, _) = (chord, toNotes (getChordRoot chord) (S.delete (intervalFrom Perfect 5) (chordToIntervals chord)))
+  where
+    toNotes :: Note -> Set Interval -> [Note]
+    toNotes root intSet = flip jumpIntervalFromNote root <$> toList intSet
+
+
+extend1 :: ExpChord -> ExpChord
+extend1 (chord, _) =  (newChord, chordToNotes newChord)
+  where
+    newChord = chordFrom (getChordRoot chord) (C.getQuality chord)
+               (extendHighestNat (getHighestNatural chord))
+               (getExtensions chord) (getSus chord)
+    extendHighestNat :: HighestNatural -> HighestNatural
+    extendHighestNat highNat =
+      if isMajor highNat then
+        majorNatural (getDegree highNat + 2)
+      else
+        nonMajorNatural (getDegree highNat + 2)
```

Comment:
This is a bit repetitive; all of the code between the two branches is common other than `majorNatural`  and `nonMajorNatural`. Maybe something like this would be a bit cleaner:

```suggestion
    extendHighestNat :: HighestNatural -> HighestNatural
    extendHighestNat highNat =
      (if isMajor highNat then majorNatural else nonMajorNatural) $ getDegree highNat + 2
```